mindmeld/active_learning/data_loading.py

```diff
@@ -0,0 +1,439 @@
+        """
+        domain_to_intents = {}
+        for domain in query_tree:
+            domain_to_intents[domain] = list(query_tree[domain])
+        return domain_to_intents
+
+    @staticmethod
+    def _get_domain_mappings(domain_to_intents: Dict) -> Dict:
+        """Creates a dictionary that maps domains to encoded ids.
+
+        Args:
+            domain_to_intents (dict): Dict mapping domains to a list of intents.
+
+        Returns:
+            domain2id (dict): dict with domain to id mappings.
+        """
+        domain2id = {}
+        domains = list(domain_to_intents)
+        for index, domain in enumerate(domains):
+            domain2id[domain] = index
+        return domain2id
+
+    @staticmethod
+    def _get_intent_mappings(domain_to_intents: Dict) -> Dict:
+        """Creates a dictionary that maps intents to encoded ids.
+
+        Args:
+            domain_to_intents (dict): Dict mapping domains to a list of intents.
+
+        Returns:
+            domain_to_intent2id (dict): dict with intent to id mappings.
+        """
+        domain_to_intent2id = {}
+        for domain in domain_to_intents:
+            intent_labels = {}
+            for index, intent in enumerate(domain_to_intents[domain]):
+                intent_labels[intent] = index
+            domain_to_intent2id[domain] = intent_labels
+        return domain_to_intent2id
+
+    @staticmethod
+    def _reverse_dict(dictionary: Dict[str, int]):
+        """
+        Returns:
+            reversed_dict (dict): Reversed dictionary.
+        """
+        reversed_dict = {v: k for k, v in dictionary.items()}
+        return reversed_dict
+
+    @staticmethod
+    def _reverse_nested_dict(dictionary: Dict[str, Dict[str, int]]):
+        """
+        Returns:
+            reversed_dict (dict): Reversed dictionary.
+        """
+        reversed_dict = {}
+
+        for parent_key, parent_value in dictionary.items():
+            reversed_dict[parent_key] = LabelMap._reverse_dict(parent_value)
+        return reversed_dict
+
+    @staticmethod
+    def get_class_labels(
+        tuning_level: str, query_list: ProcessedQueryList
+    ) -> List[str]:
+        """Creates a class label for a set of queries. These labels are used to split
+            queries by type. Labels follow the format of "domain" or "domain|intent".
+            For example, "date|get_date".
+
+        Args:
+            tuning_level (str): The hierarchy level to tune ("domain" or "intent")
+            query_list (ProcessedQueryList): Data structure containing a list of processed queries.
+        Returns:
+            class_labels (List[str]): list of labels for classification task.
+        """
+        if tuning_level == TUNE_LEVEL_DOMAIN:
+            return [f"{d}" for d in query_list.domains()]
+        elif tuning_level == TUNE_LEVEL_INTENT:
+            return [
+                f"{d}|{i}" for d, i in zip(query_list.domains(), query_list.intents())
```

Comment:
Elsewhere in mindmeld, e.g. allowed_intents we use `domain.intent` format. Have you considered that here? This notation could be confused with entity/role labels.

Reply:
Makes sense to make this consistent with allowed_intents. This list is just used to create unique strings for each intent so in terms of the function's requirements it is okay for us to switch to "." instead of "|". 

`f"{d}.{i}" for d, i in zip(query_list.domains(), query_list.intents())`